eth/executionclient/execution_client.go

```diff
@@ -117,6 +124,66 @@ func (ec *ExecutionClient) FetchHistoricalLogs(ctx context.Context, fromBlock ui
 	return logs, lastBlock, nil
 }
 
+type FetchedLogs struct {
+	logs      []ethtypes.Log
+	lastBlock uint64
+	err       error
+}
+
+// Calls FilterLogs multiple times and batches results to avoid fetching enormous amount of events
+func (ec *ExecutionClient) batchedFilterLogs(ctx context.Context, client *ethclient.Client, query ethereum.FilterQuery) <-chan FetchedLogs {
```

Comment:
We need to think if we need to test this particular method or testing FetchHistoricalLogs is enough